x-pack/plugins/security_solution/public/management/cypress/tasks/login.ts

```diff
@@ -5,242 +5,107 @@
+import type { SecurityTestUser } from '../common/constants';
+import { COMMON_API_HEADERS, request } from './common';
+
+export const ROLE = Object.freeze<Record<SecurityTestUser, SecurityTestUser>>({
+  ...ENDPOINT_SECURITY_ROLE_NAMES,
+  elastic: 'elastic',
+  elastic_serverless: 'elastic_serverless',
+  system_indices_superuser: 'system_indices_superuser',
+});
+
+interface CyLoginTask {
+  (user?: SecurityTestUser): ReturnType<typeof sendApiLoginRequest>;
+
+  /**
+   * Login using any username/password
+   * @param username
+   * @param password
+   */
+  with(username: string, password: string): ReturnType<typeof sendApiLoginRequest>;
+
+  /**
+   * Creates the provided role in kibana/ES along with a respective user (same name as role)
+   * and then login with this new user
+   * @param role
+   */
+  withCustomRole(role: Role): ReturnType<typeof sendApiLoginRequest>;
 }
 
-export const rolesMapping: { [key in ROLE]: Omit<Role, 'name'> } = {
-  t1Analyst: getT1Analyst(),
-  t2Analyst: getT2Analyst(),
-  hunter: getHunter(),
-  threatIntelligenceAnalyst: getThreatIntelligenceAnalyst(),
-  socManager: getSocManager(),
-  platformEngineer: getPlatformEngineer(),
-  endpointOperationsAnalyst: getEndpointOperationsAnalyst(),
-  endpointSecurityPolicyManager: getEndpointSecurityPolicyManager(),
-  detectionsEngineer: getDetectionsEngineer(),
-  endpointResponseActionsAccess: getWithResponseActionsRole(),
-  endpointResponseActionsNoAccess: getNoResponseActionsRole(),
-  endpointSecurityPolicyManagementRead: getEndpointSecurityPolicyManagementReadRole(),
-};
-/**
- * Credentials in the `kibana.dev.yml` config file will be used to authenticate
- * with Kibana when credentials are not provided via environment variables
- */
-const KIBANA_DEV_YML_PATH = '../../../config/kibana.dev.yml';
-
-/**
- * The configuration path in `kibana.dev.yml` to the username to be used when
- * authenticating with Kibana.
- */
-const ELASTICSEARCH_USERNAME_CONFIG_PATH = 'config.elasticsearch.username';
-
-/**
- * The configuration path in `kibana.dev.yml` to the password to be used when
- * authenticating with Kibana.
- */
-const ELASTICSEARCH_PASSWORD_CONFIG_PATH = 'config.elasticsearch.password';
-
-/**
- * The `CYPRESS_ELASTICSEARCH_USERNAME` environment variable specifies the
- * username to be used when authenticating with Kibana
- */
-const ELASTICSEARCH_USERNAME = 'ELASTICSEARCH_USERNAME';
-
 /**
- * The `CYPRESS_ELASTICSEARCH_PASSWORD` environment variable specifies the
- * username to be used when authenticating with Kibana
+ * Login to Kibana using API (not login page).
+ * By default, user will be logged in using `KIBANA_USERNAME` and `KIBANA_PASSWORD` retrieved from
+ * the cypress `env`
+ *
+ * @param user
  */
-const ELASTICSEARCH_PASSWORD = 'ELASTICSEARCH_PASSWORD';
-
-const KIBANA_USERNAME = 'KIBANA_USERNAME';
-const KIBANA_PASSWORD = 'KIBANA_PASSWORD';
+export const login: CyLoginTask = (
+  // FIXME:PT default user to `soc_manager`
```

Comment:
Q: Is this something for later?

Reply:
yes. I did not change it just yet in order to limit the amount of refactor - just in case that changing it to `soc_manager` requires test changes